src/modules/processing/services/instagram.js

```diff
@@ -60,24 +87,40 @@ async function request(url, cookie, method = 'GET', requestData) {
     return data.json();
 }
 
+async function requestMobileApi(id, cookie) {
+    const oembedURL = new URL('https://i.instagram.com/api/v1/oembed/');
+    oembedURL.searchParams.set('url', `https://www.instagram.com/p/${id}/`);
+
+    const oembed = await fetch(oembedURL, {
+        headers: {
+            ...mobileHeaders,
+            cookie
+        }
+    }).then(r => r.text());
+
+    let parsedOembed;
+    try {
+        parsedOembed = JSON.parse(oembed)
+    } catch {
+        return false;
+    }
```

Comment:
r.json()?

Reply:
> oembed endpoint sometimes returns "Private media" in plain text.

so what difference does that make with `response.json()` vs `response.text() + JSON.parse()`

Reply:
> i could wrap fetch into try {} catch {} instead of this, though

i would just do `.catch(() => {})` like everywhere else

Reply:
it throws an error and stops the sequence of (fallback) functions

Reply:
yea if it's uncaught